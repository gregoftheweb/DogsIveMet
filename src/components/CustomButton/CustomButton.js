import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  Pressable,
  useWindowDimensions,
} from 'react-native';
import {textTreatment, colorPalette} from '../../../assets/styles/globalStyles';

const CustomButton = ({onPress, text, type = 'PRIMARY', bgColor, fgColor}) => {
  const {width} = useWindowDimensions();
  return (
    <Pressable
      onPress={onPress}
      style={[
        styles.container,
        styles[`container_${type}`],
        bgColor ? {backgroundColor: bgColor} : {},
      ]}>
      <Text
        style={[
          styles.text,
          styles[`text_${type}`],
          fgColor ? {color: fgColor} : {},
        ]}>
        {text}
      </Text>
    </Pressable>
  );
};

export default CustomButton;

const styles = StyleSheet.create({
  container: {
    width: '80%',
    padding: 15,
    marginVertical: 5,
    alignItems: 'center',
    borderRadius: 5,
  },
  container_PRIMARY: {
    backgroundColor: '#00749c',
    fontWeight: '500',
  },
  container_SECONDARY: {
    borderColor: '#3b71f3',
    borderWidth: 2,
  },
  container_TERTIARY: {},
  text: {
    fontWeight: '600',
    color: 'white',
  },
  container_SMALL: {
    padding: 5,
    marginVertical: 2,
  },
  text_PRIMARY: {
    color: '#52082b',
    fontWeight: 'bold',
    fontFamily: 'sans-serif',
    fontSize: 25,
  },
  text_SECONDARY: {
    color: '#3b71f3',
  },
  text_TERTIARY: {
    color: 'gray',
  },
});
