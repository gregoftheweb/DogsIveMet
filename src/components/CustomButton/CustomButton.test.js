import React from 'react';
import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import CustomButton from './CustomButton';

it('renders correctly', () => {
  renderer.create(<CustomButton />);
});

describe('CustomButton', () => {
  test('render button with text', () => {
    const component = renderer.create(<CustomButton />);

    // Expect the result
    expect(component.toJSON()).toMatchSnapshot();
  });
});
