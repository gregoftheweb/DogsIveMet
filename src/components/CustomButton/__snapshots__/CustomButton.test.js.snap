// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton render button with text 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "marginVertical": 5,
        "padding": 15,
        "width": "80%",
      },
      Object {
        "width": "80%",
      },
      Object {
        "backgroundColor": "#00749c",
        "fontWeight": "500",
      },
      Object {},
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#00749c",
          "fontWeight": "600",
        },
        Object {
          "color": "#52082b",
          "fontFamily": "sans-serif",
          "fontSize": 25,
          "fontWeight": "bold",
        },
        Object {},
      ]
    }
  />
</View>
`;
